src/System.CommandLine.NamingConventionBinder/CommandResultExtensions.cs

```diff
@@ -46,21 +46,20 @@ internal static class CommandResultExtensions
 
         for (var i = 0; i < options.Count; i++)
         {
-            if (options[i] is Option option)
+            Option option = options[i];
```

Comment:
The `TryGetValueForOption` method looks like it isn't used by `System.CommandLine` at this point, only via  `System.CommandLine.NamingConventionBinder`, if you wanted to move it.